src/mono/wasm/testassets/WasmBasicTestApp/App/MemoryTest.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Runtime.InteropServices.JavaScript;
+
+public partial class MemoryTest
+{
+    [JSImport("joinStringArray", "main.js")]
+    internal static partial string JoinStringArray(string[] testArray);
+
+    [JSExport]
+    internal static void Run()
+    {
+        // Allocate a 2GB space (20 int arrays of 100MB, 100MB = 4 * 1024 * 1024 * 25)
+        const int arrayCnt = 20;
+        int[][] arrayHolder = new int[arrayCnt][];
+        string errors = "";
+        for (int i = 0; i < arrayCnt; i++)
+        {
+            try
+            {
+                arrayHolder[i] = new int[1024 * 1024 * 25];
+            }
+            catch (Exception ex)
+            {
+                errors += $"Exception {ex} was thrown on i={i}";
+            }
+        }
+
+        // marshall string array to JS
+        string [] testArray = new [] { "M", "e", "m", "o", "r", "y", "T", "e", "s", "t" };
```

Comment:
Please make each string here random (constants are not allocated) and large enough, so that the JS side is forced to make allocations above 2GB mark. And overflow to negative numbers. Maybe run this in a loop.

Reply:
Then let's use the `arrayHolder` that is already prepared.

Reply:
2GB string allocation with random data is taking too long, not only to run WBT but to even debug it.

Reply:
Also, we cannot just take `arrayHolder` and send it to JS to force JS to allocate 2GB because interop does not support array of arrays. Allocating flat array of 2GB requires more linear memory and can be problematic from other reasons than investigated in the issue. Allocating an array of strings theoretically should work but as I already mentioned, it takes tens of minutes to finish the allocation, even using `StringBuilder`.
@maraf, do you have a good idea how to fulfill the requirements?

Reply:
Never mind, I realized that the key point was "allocations above 2GB mark", not "allocations of 2GB".

Reply:
We're allocating `2 621 440 000 bytes` so more than max int. There are no traces of overflow. Let's merge this to take care of the relink fix PR that is built on top of this.